sootup.core/src/main/java/sootup/core/util/printer/JimplePrinter.java

```diff
@@ -196,8 +191,12 @@ public void printTo(SootClass<?> cl, PrintWriter out, LabeledStmtPrinter printer
           SootField f = (SootField) fieldIt.next();
           printer.newline();
           printer.handleIndent();
-          printer.literal(f.getDeclaration());
-          printer.literal(";");
+          if (!f.getModifiers().isEmpty()) {
+            printer.literal(FieldModifier.toString(f.getModifiers()));
```

Comment:
Offtopic: FieldModifier.toString(...) already assumes valid Modifier combinations (theoretically its currently possible to assign public+private+protected modifier at the same time) - maybe we should implement a wrapper of EnumSet<...Modifier> to handle only valid combinations and implement toString there?